velox/vector/ComplexVector.cpp

```diff
@@ -673,7 +673,7 @@ void RowVector::resize(vector_size_t newSize, bool setNotNull) {
       // to skip uniqueness check since effectively we are just changing
       // the length.
       if (newSize > oldSize) {
-        VELOX_CHECK(child.unique(), "Resizing shared child vector");
+        // VELOX_CHECK(child.unique(), "Resizing shared child vector");
```

Comment:
Why have you commented this line ?

Reply:
This check makes sure if the vectors are shared we don't resize them. I will try an alternate approach like you and @Yuhta suggested.